src/apps/btc/btc_common.c

```diff
@@ -738,15 +772,16 @@ bool btc_common_multisig_hash(
         rust_sha256_update(ctx, &num, sizeof(num));
     }
     { // 5. xpubs
+        const XPub* xpubs[sizeof(multisig->xpubs)] = {0};
+        for (size_t i = 0; i < multisig->xpubs_count; i++) {
+            xpubs[i] = &multisig->xpubs[i];
+        }
+        if (sort_xpubs) {
+            qsort(xpubs, multisig->xpubs_count, sizeof(*xpubs), _xpubs_sort_comp);
```

Comment:
Feels like serializing first, then sorting and finally sha256 would work just as well if not better since you don't need to serialize twice? Assuming sorting would work without ignoring the serialized xpub version.

Reply:
Yeah I guess. I'll try.

> Assuming sorting would work without ignoring the serialized xpub version.

That question seems independent?

Reply:
Done (serialize, then sort + hash).

Reply:
:+1: